scipy/stats/tests/test_multicomp.py

```diff
@@ -250,6 +274,94 @@ def test_alternatives(self, alternative, statistic, pvalue):
             )
             assert_allclose(res.pvalue, pvalue, atol=1e-7)
 
+    @pytest.mark.parametrize(
+        'samples, control, alternative, ci_low, ci_high',
+        [
+            (
+                samples_1, control_1, 'two-sided',
+                *ci_1_twosided
+             ),
+            (
+                samples_1, control_1, 'less',
+                ci_1_less,
+                [np.inf, np.inf, np.inf, np.inf, np.inf]
+            ),
+            (
+                samples_1, control_1, 'greater',
+                [-np.inf, -np.inf, -np.inf, -np.inf, -np.inf],
+                ci_1_greater
+            ),
+            (
+                samples_3, control_3, 'two-sided',
+                *ci_3_twosided
+             ),
+            (
+                samples_3, control_3, 'less',
+                ci_3_less,
+                [np.inf, np.inf, np.inf]
+            ),
+            (
+                samples_3, control_3, 'greater',
+                [-np.inf, -np.inf, -np.inf],
+                ci_3_greater
+            )
+        ]
+    )
+    def test_confidence_interval(
+        self, samples, control, alternative, ci_low, ci_high
+    ):
+        rng = np.random.default_rng(189117774084579816190295271136455278291)
+
+        res = stats.dunnett(
+            *samples, control=control, alternative=alternative,
+            random_state=rng
+        )
+
+        assert res._ci is None
+        assert res._ci_cl is None
+        ci = res.confidence_interval(confidence_level=0.95)
+        assert_allclose(ci.low, ci_low, atol=1e-2)
+        assert_allclose(ci.high, ci_high, atol=1e-2)
+
+        # re-run to use the cached value "is" to check id as same object
+        assert res._ci is ci
+        assert res._ci_cl == 0.95
+        ci_ = res.confidence_interval(confidence_level=0.95)
+        assert ci_ is ci
+
+    def toto(self):
```

Comment:
I know 😉 Fixing this and using the new structure instead of redefining all the time the samples and control